lib/contracts/formatters.rb

```diff
@@ -85,8 +85,28 @@ def plain?
 
       def useful_to_s?
         # Useless to_s value or no custom to_s behavious defined
-        # Ruby < 2.0 makes inspect call to_s so this won't work
-        @value.to_s != "" && @value.to_s != @value.inspect
+        !empty_to_s? && custom_to_s?
+      end
+
+      def empty_to_s?
+        @value.to_s.empty?
+      end
+
+      def custom_to_s?
+        if @value.class == Class # It's a class contract
```

Comment:
Don't really know what is better in that case. So for you @sfcgeorge to decide :)